be/src/exec/olap_scan_node.cpp

```diff
@@ -681,7 +681,11 @@ Status OlapScanNode::start_scan_thread(RuntimeState* state) {
             }
             OlapScanner* scanner = new OlapScanner(
                 state, this, _olap_scan_node.is_preaggregation, _need_agg_finalize, *scan_range, scanner_ranges);
+            // add scanner to pool before doing prepare.
+            // so that scanner can be automatically deconstructed if prepare failed.
             _scanner_pool->add(scanner);
```

Comment:
It doesn't matter whether you add the scanner to the pool before or after doing prepare.
if you put scanner preparing before add the scanner_pool，the scanner will be decontruced.

Reply:
Why? the `scanner` here is created by `new` operator. How can it be deconstructed automatically?